rasa/nlu/data_augmentation/augmenter.py

```diff
@@ -0,0 +1,638 @@
+            "accuracy": intent_report["accuracy"],
+        }
+
+        intent_report["accuracy"] = acc_dict
+        intent_summary["accuracy"] = {"accuracy_change": accuracy_change}
+
+    for intent in (
+        intents_to_augment
+        | changed_intents
+        | {"micro avg", "macro avg", "weighted avg"}
+    ):
+        if intent not in classification_report:
+            continue
+
+        intent_results_original = classification_report[intent]
+        intent_results = intent_report[intent]
+
+        # Record performance changes for augmentation based on the diversity criterion
+        precision_change = (
+            intent_results["precision"] - intent_results_original["precision"]
+        )
+        recall_change = intent_results["recall"] - intent_results_original["recall"]
+        f1_change = intent_results["f1-score"] - intent_results_original["f1-score"]
+
+        intent_results["precision_change"] = intent_summary[intent][
+            "precision_change"
+        ] = precision_change
+        intent_results["recall_change"] = intent_summary[intent][
+            "recall_change"
+        ] = recall_change
+        intent_results["f1-score_change"] = intent_summary[intent][
+            "f1-score_change"
+        ] = f1_change
+        intent_report[intent] = intent_results
+
+    return (intent_summary, intent_report)
+
+
+def _create_summary_report(
+    intent_report: Dict[Text, float],
+    classification_report: Dict[Text, Dict[Text, Any]],
+    training_intents: List[Text],
+    pooled_intents: Set[Text],
+    output_directory: Text,
+) -> Tuple[Dict[Text, Dict[Text, float]], Set[Text]]:
+    """Creates a summary of the effect of data augmentation.
+
+    Args:
+        intent_report: Report of the model run *with* data augmentation.
+        classification_report: Classification report of the model run *without* data augmentation.
+        training_intents: All intents in the training data (non-augmented).
+        pooled_intents: The intents that have been selected for data augmentation.
+        output_directory: Directory to store the output reports in.
+
+    Returns:
+        Tuple representing the data augmentation summary as well as the set of changed intents.
+    """
+
+    # Retrieve intents for which performance has changed
+    changed_intents = (
+        _get_intents_with_performance_changes(
+            classification_report, intent_report, training_intents,
+        )
+        - pooled_intents
+    )
+
+    # Create and update result reports
+    report_tuple = _create_augmentation_summary(
+        pooled_intents, changed_intents, classification_report, intent_report,
+    )
+
+    intent_summary = report_tuple[0]
+    intent_report.update(report_tuple[1])
+
+    # Store report to file
+    rasa.shared.utils.io.dump_obj_as_json_to_file(
+        os.path.join(output_directory, "intent_report.json"), intent_report,
+    )
+
+    return intent_summary, changed_intents
```

Comment:
I see that the only call to `_create_summary_report` is not using `changed_intents`. i think we don't need to return it.